javatests/arcs/core/data/RecipeTest.kt

```diff
@@ -194,18 +214,19 @@ class RecipeTest {
     fun recipeToPlan_fullExample() {
 
         val peopleStorageKey = "reference-mode://{db://abcd@arcs/Person}{db://abcd@arcs//handle/people}"
-
         val peopleHandle = Recipe.Handle(
             name = "people",
             fate = Fate.MAP,
             type = personCollectionType,
             storageKey = peopleStorageKey
         )
 
+        val contactsStorageKey = "create://contacts"
```

Comment:
IIRC we needed the CreatableStorageKey because we didn't have Handle objects, only handle connections. Do we still need them now?

Reply:
Yes, we could replace creatable storage keys with handles without storage keys. To do this we need to take another step - have HandleConnections reference a Handle, and not its storage key directly. This has been pulled out of this PR because of the size of refactoring needed.

Reply:
yep, i didn't mean this PR, just maybe a tracking bug or todo? i don't think we should keep both mechanisms in general, unless there is a need. thanks!